File: cpp/source/Device.cpp
#include "Device.hpp"

#include "state/device/Disconnected.hpp"

using namespace fm;
using namespace fm::state;

Device::Device(Listener& listener, core::https::IHttpsClient& coreClient, const std::string& _certPath) :
    IClient(listener, coreClient, _certPath)
{
    setState(std::make_unique<device::Disconnected>(*this, listener, *backend.get()));
}

std::string Device::toString() const
{
    return "Device";
}

File: java/src/main/java/com/fleetmgr/client/sdk/Device.java
package com.fleetmgr.client.sdk;

import com.fleetmgr.client.sdk.state.device.Disconnected;

import java.util.concurrent.ExecutorService;

/**
 * Created by: Bartosz Nawrot
 * Date: 04.09.2018
 * Description:
 */
public class Device extends Client {

    public Device(String coreAddress, String key,
                  Listener listener, ExecutorService executor) {
        super(coreAddress, key, listener, executor);
        setState(new Disconnected(key,this, coreClient, listener, executor));
    }
}
